src/Scorer.js

```diff
@@ -0,0 +1,93 @@
+function Scorer () {
+  this.runningTotals = []
+  this._rolls = []
+  this._scoringIndex = 0
+}
+
+Scorer.prototype = {
+  constructor: Scorer,
+
+  addRoll: function (roll) {
+    var updateScores
+    this._rolls.push(roll)
+    do {
+      updateScores = this._tryUpdatingScores()
+    } while (updateScores)
+  },
+
+  total: function () {
+    return this._lastRunningTotal()
+  },
+
+  _tryUpdatingScores: function () {
+    var i = this._scoringIndex
+    if (this._scoredFrames() === 10) {
+      return false
+    }
+    if (this._rolls[i] === 10) {
```

Comment:
would it be better to have a frame object here which could store whether it's a strike or spare? so you would do `if (this.isStrike()) {}` or something? problem i had with that was with wanting to be able to store the rolls in an array so i could iterate over them easily, and not wanting to store them in the frame as well... so you could have them stored in an array in `scorer`, and then... store an index reference to the appropriate rolls in the `frame`? thoughts...